File: create_dirs.py
#!/usr/bin/env python
"""Create Next.js project directory structure"""

import os

basePath = r'C:\Users\DELL\Documents\projet\immo\frontend'
dirs = [
    r'src\app\catalogue',
    r'src\app\maison\[id]',
    r'src\app\a-propos',
    r'src\app\admin\login',
    r'src\app\admin\dashboard',
    r'src\app\admin\maisons\nouvelle',
    r'src\app\admin\maisons\[id]',
    r'src\components\layout',
    r'src\components\maison',
    r'src\components\ui',
    r'src\lib',
    r'src\types',
    r'src\providers',
    r'public'
]

created_count = 0
for dir_path in dirs:
    full_path = os.path.join(basePath, dir_path)
    try:
        os.makedirs(full_path, exist_ok=True)
        created_count += 1
        print(f"✓ Created: {dir_path}")
    except Exception as e:
        print(f"✗ Failed to create {dir_path}: {e}")

print(f"\n✓ Successfully created {created_count}/{len(dirs)} directories")

File: create_dirs.js
const fs = require('fs');
const path = require('path');

const dirs = [
  'frontend/src/app/catalogue',
  'frontend/src/app/maison/[id]',
  'frontend/src/app/a-propos',
  'frontend/src/app/admin/login',
  'frontend/src/app/admin/dashboard',
  'frontend/src/app/admin/maisons/nouvelle',
  'frontend/src/app/admin/maisons/[id]',
  'frontend/src/components/layout',
  'frontend/src/components/maison',
  'frontend/src/components/ui',
  'frontend/src/lib',
  'frontend/src/types',
  'frontend/src/providers',
  'frontend/public'
];

const baseDir = 'C:\\Users\\DELL\\Documents\\projet\\immo';

dirs.forEach(dir => {
  const fullPath = path.join(baseDir, dir);
  try {
    fs.mkdirSync(fullPath, { recursive: true });
    console.log(`✓ Created: ${fullPath}`);
  } catch (err) {
    console.error(`✗ Error creating ${fullPath}:`, err.message);
  }
});

console.log('\n✓ Directory structure created successfully!');
